oak/server/rust/oak_loader/src/main.rs

```diff
@@ -67,34 +74,26 @@ fn read_file(filename: &str) -> Result<Vec<u8>, Box<dyn std::error::Error>> {
     Ok(buffer)
 }
 
-fn start_runtime(
-    application: String,
-    runtime_config: oak_runtime::RuntimeConfiguration,
-) -> Result<(), Box<dyn std::error::Error>> {
-    info!("Loading Oak Runtime");
-
-    let app_config = {
-        let buffer = read_file(&application)?;
-        ApplicationConfiguration::decode(&buffer[..])
-            .map_err(|error| format!("Failed to decode application configuration: {:?}", error))?
-    };
-
-    // Start the Runtime from the given config.
-    let (_runtime, _initial_handle) = configure_and_run(app_config, runtime_config)
-        .map_err(|error| format!("Runtime error: {:?}", error))?;
-
-    // The Runtime kicks off its own threads for the initial Node and any subsequently created
-    // Nodes, so just park the current thread.
-    thread::park();
-    Ok(())
-}
-
 fn main() {
-    env_logger::init();
+    simple_logger::init().expect("failed to initialize logger");
     let opt = Opt::from_args();
-    let application = opt.application.clone();
 
-    // Start the runtime in a new thread
+    let app_config_data = read_file(&opt.application).unwrap_or_else(|err| {
+        error!(
```

Comment:
Is it worth returning `Result` from main and just use `?` to propagate errors up (which will panic and log)?

Reply:
Neat, don't think I knew you could do that.